model/data_retention_policy.go

```diff
@@ -8,20 +8,101 @@ import (
 	"io"
 )
 
-type DataRetentionPolicy struct {
+type GlobalRetentionPolicy struct {
 	MessageDeletionEnabled bool  `json:"message_deletion_enabled"`
 	FileDeletionEnabled    bool  `json:"file_deletion_enabled"`
 	MessageRetentionCutoff int64 `json:"message_retention_cutoff"`
 	FileRetentionCutoff    int64 `json:"file_retention_cutoff"`
 }
 
-func (drp *DataRetentionPolicy) ToJson() string {
-	b, _ := json.Marshal(drp)
+type RetentionPolicy struct {
+	Id           string `json:"id"`
+	DisplayName  string `json:"display_name"`
+	PostDuration int64  `json:"post_duration"`
+}
+
+type ChannelDisplayInfo struct {
+	Id              string `json:"id"`
+	DisplayName     string `json:"display_name"`
+	TeamDisplayName string `json:"team_display_name"`
+}
+
+type TeamDisplayInfo struct {
+	Id          string `json:"id"`
+	DisplayName string `json:"display_name"`
+}
+
+type RetentionPolicyEnriched struct {
+	Id           string               `json:"id"`
+	DisplayName  string               `json:"display_name"`
+	PostDuration int64                `json:"post_duration"`
+	Teams        []TeamDisplayInfo    `json:"teams"`
+	Channels     []ChannelDisplayInfo `json:"channels"`
+}
```

Comment:
Go has a construct called Composition instead of inheritance, but work similar. 
```
type RetentionPolicyEnriched struct {
	RetentionPolicy
	Teams        []TeamDisplayInfo    `json:"teams"`
	Channels     []ChannelDisplayInfo `json:"channels"`
}
```